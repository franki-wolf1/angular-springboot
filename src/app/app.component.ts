import { Component } from '@angular/core';
import { RouterOutlet } from '@angular/router'; 
import { MaterialModule } from './material.module';
 
import { AngularFirestoreModule } from '@angular/fire/compat/firestore'; 
import { environment } from '../environments/emviroment';
import { AngularFireModule } from '@angular/fire/compat';

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [
    RouterOutlet, MaterialModule, 
    AngularFirestoreModule
  ],
  templateUrl: './app.component.html',
  styleUrl: './app.component.sass'
})
export class AppComponent {
  title = 'inter-ruta-passengers';

  constructor() {}

  ngOnInit() { 
    AngularFireModule.initializeApp(environment.firebase);
  }
}
