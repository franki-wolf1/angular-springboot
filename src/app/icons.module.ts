import { NgModule } from '@angular/core';

import { TablerIconsModule } from 'angular-tabler-icons';
import * as TablerIcons from 'angular-tabler-icons/icons';
import { IconCamera, IconHeart, IconBrandGithub } from 'angular-tabler-icons/icons';

// Select some icons (use an object, not an array)
const icons = {
  IconCamera,
  IconHeart,
  IconBrandGithub
};

@NgModule({
  imports: [
    TablerIconsModule.pick(TablerIcons)
  ],
  exports: [
    TablerIconsModule
  ]
})
export class IconsModule { }

// NOTES:
// 1. We add TablerIconsModule to the 'exports', since the <i-tabler> component will be used in templates of parent module
// 2. Don't forget to pick some icons using TablerIconsModule.pick({ ... })