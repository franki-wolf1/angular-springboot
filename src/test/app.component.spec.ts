import { TestBed, ComponentFixture } from '@angular/core/testing'; 
import { AngularFirestoreModule } from '@angular/fire/compat/firestore';
import { AngularFireModule } from '@angular/fire/compat';  
import { environment } from '../environments/emviroment';
import { AppComponent } from '../app/app.component';
import { MaterialModule } from '../app/material.module';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({ 
      imports: [AngularFirestoreModule, AngularFireModule, MaterialModule],
    }).compileComponents();

    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
  });

  test('should create the app', () => {
    expect(component).toBeTruthy();
  });

  test('should have a title', () => {
    expect(component.title).toBeDefined();
    expect(component.title).toBe('inter-ruta-passengers');
  });

  test('should initialize AngularFireModule with environment.firebase', () => {
    expect(() => AngularFireModule.initializeApp(environment.firebase)).not.toThrowError();
  });

  test('debe hacer mach con snapshot', ()=>{
    const fixture = TestBed.createComponent(AppComponent);
    const compiler = fixture.nativeElement as HTMLElement;
    expect( compiler ).toMatchSnapshot();
  })

  // Add more test cases as needed
});
