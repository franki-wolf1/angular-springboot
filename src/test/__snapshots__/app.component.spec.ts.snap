// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent debe hacer mach con snapshot 1`] = `
<div
  id="root4"
>
  <router-outlet />
  
</div>
`;
